src/main/java/com/nha/abdm/wrapper/hrp/controller/GatewayController.java

```diff
@@ -0,0 +1,98 @@
+package com.nha.abdm.wrapper.hrp.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.nha.abdm.wrapper.hrp.discoveryLinking.responses.ConfirmResponse;
+import com.nha.abdm.wrapper.hrp.discoveryLinking.responses.DiscoverResponse;
+import com.nha.abdm.wrapper.hrp.discoveryLinking.responses.InitResponse;
+import com.nha.abdm.wrapper.hrp.discoveryLinking.services.DiscoverLinkingService;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.services.HipInitiatedService;
+import com.nha.abdm.wrapper.hrp.properties.ApplicationConfig;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.responses.LinkRecordsResponse;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.responses.OnConfirmResponse;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.responses.OnInitResponse;
+import com.nha.abdm.wrapper.hrp.services.CommonServices;
+import com.nha.abdm.wrapper.hrp.common.Utils;
+
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+@RestController
```

Comment:
This class is not exposing any REST APIs, so the `RestController` is not needed. Also, the name of the class can be renamed something like `GatewayService`